src/windows/wslaservice/exe/WSLASession.cpp

```diff
@@ -213,15 +228,67 @@ void WSLASession::CopyDisplayName(_Out_writes_z_(bufferLength) PWSTR buffer, siz
     wcscpy_s(buffer, bufferLength, m_displayName.c_str());
 }
 
-HRESULT WSLASession::PullImage(LPCWSTR Image, const WSLA_REGISTRY_AUTHENTICATION_INFORMATION* RegistryInformation, IProgressCallback* ProgressCallback)
+HRESULT WSLASession::PullImage(LPCSTR ImageUri, const WSLA_REGISTRY_AUTHENTICATION_INFORMATION* RegistryAuthenticationInformation, IProgressCallback* ProgressCallback)
+try
 {
-    return E_NOTIMPL;
+    RETURN_HR_IF_NULL(E_POINTER, ImageUri);
+
+    std::lock_guard lock{m_lock};
+
+    ServiceProcessLauncher launcher{nerdctlPath, {nerdctlPath, "pull", ImageUri}};
+    auto result = launcher.Launch(*m_virtualMachine.Get()).WaitAndCaptureOutput();
+
+    RETURN_HR_IF_MSG(E_FAIL, result.Code != 0, "Pull image failed: %hs", launcher.FormatResult(result).c_str());
+
+    return S_OK;
 }
+CATCH_RETURN();
 
-HRESULT WSLASession::ImportImage(ULONG Handle, LPCWSTR Image, IProgressCallback* ProgressCallback)
+HRESULT WSLASession::LoadImage(ULONG ImageHandle, IProgressCallback* ProgressCallback)
+try
 {
-    return E_NOTIMPL;
+    HANDLE imageFileHandle = ULongToHandle(ImageHandle);
```

Comment:
The handle needs to opened in the current process to read from it. 

We can do that with: 

```
auto localHandle = wsl::windows::common::wslutil::DuplicateHandleFromCallingProcess(ImageHandle);
```